src/components/policy/policy_regular/src/policy_table/types.cc

```diff
@@ -1539,6 +1608,417 @@ void DeviceParams::ReportErrors(rpc::ValidationReport* report__) const {
   }
 }
 
+// VehicleDataItem methods
+VehicleDataItem::VehicleDataItem() : CompositeType(kUninitialized) {}
+
+VehicleDataItem::VehicleDataItem(const VehicleDataItem& vehicle_data)
+    : CompositeType(vehicle_data.initialization_state__)
+    , name(vehicle_data.name)
+    , type(vehicle_data.type)
+    , key(vehicle_data.key)
+    , mandatory(vehicle_data.mandatory)
+    , params(vehicle_data.params)
+    , array(vehicle_data.array)
+    , since(vehicle_data.since)
+    , until(vehicle_data.until)
+    , removed(vehicle_data.removed)
+    , deprecated(vehicle_data.deprecated)
+    , minvalue(vehicle_data.minvalue)
+    , maxvalue(vehicle_data.maxvalue)
+    , minsize(vehicle_data.minsize)
+    , maxsize(vehicle_data.maxsize)
+    , minlength(vehicle_data.minlength)
+    , maxlength(vehicle_data.maxlength) {}
+
+VehicleDataItem::VehicleDataItem(const Json::Value* value__)
+    : CompositeType(InitHelper(value__, &Json::Value::isObject))
+    , name(impl::ValueMember(value__, "name"))
+    , type(impl::ValueMember(value__, "type"))
+    , key(impl::ValueMember(value__, "key"))
+    , mandatory(impl::ValueMember(value__, "mandatory"))
+    , params(impl::ValueMember(value__, "params"))
+    , array(impl::ValueMember(value__, "array"))
+    , since(impl::ValueMember(value__, "since"))
+    , until(impl::ValueMember(value__, "until"))
+    , removed(impl::ValueMember(value__, "removed"))
+    , deprecated(impl::ValueMember(value__, "deprecated"))
+    , minvalue(impl::ValueMember(value__, "minvalue"))
+    , maxvalue(impl::ValueMember(value__, "maxvalue"))
+    , minsize(impl::ValueMember(value__, "minsize"))
+    , maxsize(impl::ValueMember(value__, "maxsize"))
+    , minlength(impl::ValueMember(value__, "minlength"))
+    , maxlength(impl::ValueMember(value__, "maxlength")) {}
+
+VehicleDataItem::~VehicleDataItem() {}
+
+const std::string VehicleDataItem::kInteger = "Integer";
+const std::string VehicleDataItem::kStruct = "Struct";
+const std::string VehicleDataItem::kString = "String";
+const std::string VehicleDataItem::kFloat = "Float";
+const std::string VehicleDataItem::kBoolean = "Boolean";
+const std::string VehicleDataItem::kName = "name";
```

Comment:
Is this used anywhere?

Reply:
@jacobkeeler removed in 76277de2068688672613bec5ac92a1beae5574dd